src/arrow2/src/compute/zip.rs

```diff
@@ -0,0 +1,174 @@
+use crate::{
+    array::{Array, BooleanArray, MutableArray, MutableBooleanArray, MutablePrimitiveArray},
+    bitmap::Bitmap,
+    datatypes::DataType,
+};
+
+/// Zip two arrays by some boolean mask. Where the mask evaluates `true` values of `true_values`
+/// are taken, where the mask evaluates `false` values of `false_values` are taken.
+///
+/// # Arguments
+/// * `mask` - Boolean values used to determine from which array to take the values.
+/// * `true_values` - Values of this array are taken if mask evaluates `true`
+/// * `false_values` - Values of this array are taken if mask evaluates `false`
+pub fn zip(
```

Comment:
This seems to do the same thing as our `if_else` kernels that you can find here
https://github.com/Eventual-Inc/Daft/blob/75ad85a9eeee13603cfc5a158d87a1d141e785cc/src/daft-core/src/array/ops/if_else.rs#L119

Reply:
series method:

https://github.com/Eventual-Inc/Daft/blob/75ad85a9eeee13603cfc5a158d87a1d141e785cc/src/daft-core/src/series/ops/if_else.rs#L7

Reply:
So i just tried using `if_else`, but that appears to implicitly cast to supertypes. Some dialects (such as duckdb) require explicit casting, 

```sql
select coalesce(null, '1', 1);
Binder Error: Cannot mix values of type VARCHAR and INTEGER_LITERAL in COALESCE operator - an explicit cast is required
```

but other systems such as datafusion allow implicit casting here

```sql
> select coalesce(null, '1', 1);
+-----------------------------------+
| coalesce(NULL,Utf8("1"),Int64(1)) |
+-----------------------------------+
| 1                                 |
+-----------------------------------+
1 row(s) fetched.
Elapsed 0.035 seconds.
```